### DISCUSSION TITLE
[SUPPORT]  Auto-clean causes Firefox to log out of all plugins requiring logins

### DISCUSSION THREAD

**Discussion by StrangeRanger** [Cookie-AutoDelete/Cookie-AutoDelete#984] [Q&A / Support] [answered]
**Please describe what you need support with.** 
I'm not sure if this relates to local storage as mentioned [in the docs](https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/wiki/Documentation#uncheck-keep-localstorage-on-new-greylist-expressions), so I apologize if it is. 

I find that if I have auto clean enabled, after quitting Firefox, the next time I open the application, I'm logged out of all of the plugins that require logins (i.e. Grammarly and LastPass). Is this a bug, is it intentional, or do I not have 'Cookie-AutoDelete' configured properly? Is there a way to stop 'Cookie-AutoDelete' from logging me out of my plugins?

**Comment by kennethtran93** [comment]
Are those plugin webextensions as well?  Also, do you have those sites whitelisted?  I'm unsure if extensions can also store cookies and whatnot, so you may need to whitelist cookies and site data from the extension page as well.

Check the cleanup logs (if enabled). If not, enable it and then check what it cleaned up.  From the domains given for each cleanup entry, you'll need to see which ones correspond to the one you need.

Are you using Containers of any kind by chance?  If so and those plugin/sites are accessed through another container that doesn't have a whitelist, you'll need to set it up accordingly for all containers.  This is currently a browser limitation where site data other than cookies can only be cleared by hostname for ALL containers.

**Comment by StrangeRanger** [comment]
Yes, they are web extensions, or at least that's what is said in the URL.

<img width="519" alt="Screen Shot 2020-12-03 at 6 20 07 PM" src="https://user-images.githubusercontent.com/30395560/101113603-a3f64a80-35d7-11eb-9979-aa68ce69bf4c.png">

Do I need to whitelist the site the plugin is from or the extension URL?

And no, I'm not using containers.

**Comment by StrangeRanger** [comment]
And yes, I do have the last pass extension whitelisted
<img width="792" alt="Screen Shot 2020-12-03 at 6 24 21 PM" src="https://user-images.githubusercontent.com/30395560/101113700-d9029d00-35d7-11eb-84c5-80c28ca95f6c.png">

**Comment by kennethtran93** [comment]
> Yes, they are web extensions, or at least that's what is said in the URL.
> 
> <img alt="Screen Shot 2020-12-03 at 6 20 07 PM" width="519" src="https://user-images.githubusercontent.com/30395560/101113603-a3f64a80-35d7-11eb-9979-aa68ce69bf4c.png">
> 
> Do I need to whitelist the site the plugin is from or the extension URL?
> 
> And no, I'm not using containers.

I'd say all related sites to that plugin and the extension URL.

Make sure you only check the global clean type in the CAD settings ONCE, then login again.  The global settings causes all of the existing data of that site data type to be removed upon checking those boxes.

Check the cleanup logs if you can.  They'll contain cookie names and hostnames for those cookie names.  I'm unsure how each extension stores their data.

**Comment by StrangeRanger** [comment, marked as answer]
So apparently, for it to ignore the login reliant extensions/add-ons, I have to whitelist the website itself. I'm assuming this is because the cookie used to authenticate to the website is saved, and that's what get's deleted every time.

**Comment by kennethtran93** [comment]
Yes, if extension relies on cookies from websites.  We do not know if cookies originated from a website or extension.  All we do is query all cookies, and check the cookie names and urls with the whitelists/greylists.

**Comment by StrangeRanger** [comment]
Alright. Sorry for taking up your time for something such as this.

**Comment by kennethtran93** [comment]
No worries.  I'm going to move this to a discussion with marked answer.
